contrib/win32/openssh/OpenSSHUtils.psm1

```diff
@@ -0,0 +1,410 @@
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({Test-Path $_ })]
+        [string]$FilePath,
+        [ValidateNotNull()]
+        [System.Security.Principal.NTAccount[]] $Owners = $currentUser,
+        [System.Security.Principal.NTAccount[]] $AnyAccessOK,
+        [System.Security.Principal.NTAccount[]] $ReadAccessOK,
+        [System.Security.Principal.NTAccount[]] $ReadAccessNeeded,
+        [switch] $Quiet
+    )   
+    
+    Write-host "----------Validating $FilePath----------"
+    $return = Fix-FilePermissionInternal @PSBoundParameters
+
+    if($return -contains $true) 
+    {
+        #Write-host "Re-check the health of file $FilePath"
+        Fix-FilePermissionInternal @PSBoundParameters
+    }
+}
+
+<#
+    .Synopsis
+    Fix-FilePermissionInternal
+#>
+function Fix-FilePermissionInternal {
+    param (
+        [parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$FilePath,
+        [ValidateNotNull()]
+        [System.Security.Principal.NTAccount[]] $Owners = $currentUser,
+        [System.Security.Principal.NTAccount[]] $AnyAccessOK,
+        [System.Security.Principal.NTAccount[]] $ReadAccessOK,
+        [System.Security.Principal.NTAccount[]] $ReadAccessNeeded,
+        [switch] $Quiet
+    )
+
+    $acl = Get-Acl $FilePath
+    $needChange = $false
+    $health = $true
+    if ($Quiet)
+    {
+        $result = 'Y'
+    }
+    
+    if(-not $Owners.Contains([System.Security.Principal.NTAccount]$($acl.Owner)))
+    {
+        if (-not $Quiet) {
+            $warning = "Current owner: '$($acl.Owner)'. '$($Owners[0])' should own $FilePath."
+            Do {
+                Write-Warning $warning
+                $input = Read-Host -Prompt "Shall I set the file owner? [Yes] Y; [No] N (default is `"Y`")"
+                if([string]::IsNullOrEmpty($input))
+                {
+                    $input = 'Y'
+                }        
+            } until ($input -match "^(y(es)?|N(o)?)$")
+            $result = $Matches[0]
+        }        
+
+        if($result.ToLower().Startswith('y'))
+        {
+            $needChange = $true
+            $acl.SetOwner($Owners[0])
+            Write-Host "'$($Owners[0])' now owns $FilePath. " -ForegroundColor Green
+        }
+        else
+        {
+            $health = $false
+            Write-Host "The owner is still set to '$($acl.Owner)'." -ForegroundColor Yellow
+        }
+    }
+
+    $ReadAccessPerm = ([System.UInt32] [System.Security.AccessControl.FileSystemRights]::Read.value__) -bor `
+                    ([System.UInt32] [System.Security.AccessControl.FileSystemRights]::Synchronize.value__)
+    $realAnyAccessOKList = $AnyAccessOK + @($systemAccount, $adminsAccount)
+    $realReadAcessOKList = $ReadAccessOK + $ReadAccessNeeded
+    $realReadAccessNeeded = $ReadAccessNeeded
+
+    foreach($a in $acl.Access)
+    {
+        if(($realAnyAccessOKList -ne $null) -and $realAnyAccessOKList.Contains($a.IdentityReference))
```

Comment:
no, we need to ignore the indentifies in the list. otherwise, it will fall to the else if block